packages/zoltan2/src/directory/Zoltan2_Directory.hpp

```diff
@@ -194,11 +195,21 @@ class Zoltan2_Directory {
 
       // fill
       size_t cnt = 0;
+
+      for(auto itr = node_map.begin(); itr != node_map.end(); ++itr) {
+        if(itr->second.free == 0) {
+          local_gids[cnt++] = itr->first;
+        }
+      }
+
+      // TODO: Restore Kokkos unordered_map
```

Comment:
You could have just used a HostSpace version of `Kokkos::UnorderedMap`, but just using a standard library class should be fine too.

Reply:
@mhoemmen Thanks for reviewing this. Did not think of doing the host space which would have been easier. We are planning to do further development to use kokkos so eventually I'll be converting this back once I resolve some std::vector issues.